lib/spack/llnl/util/filesystem.py

```diff
@@ -974,6 +974,43 @@ def remove_linked_tree(path):
             shutil.rmtree(path, True)
 
 
+@contextmanager
+def safe_remove(*files_or_dirs):
+    """Context manager to remove the files passed as input, but restore
+    them in case any exception is raised in the context block.
+
+    Filenames clash from single files in different directories are currently
+    not handled.
```

Comment:
I'll lift this limitation while improving the function according to review.

Reply:
Removed these two lines since the limitations is not there. This is the test for the case above:

https://github.com/spack/spack/blob/9063f3b5b80a5da9afde082a2497663d64a2bae5/lib/spack/spack/test/llnl/util/filesystem.py#L523-L529